cmake/OpenCVFindWebP.cmake

```diff
@@ -21,13 +19,34 @@ else()
 
     # Look for the library.
     FIND_LIBRARY(WEBP_LIBRARY NAMES webp)
-    MARK_AS_ADVANCED(WEBP_LIBRARY)
+    FIND_LIBRARY(WEBP_MUX_LIBRARY NAMES webpmux)
+    FIND_LIBRARY(WEBP_DEMUX_LIBRARY NAMES webpdemux)
 
     # handle the QUIETLY and REQUIRED arguments and set WEBP_FOUND to TRUE if
     # all listed variables are TRUE
     INCLUDE(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
     FIND_PACKAGE_HANDLE_STANDARD_ARGS(WebP DEFAULT_MSG WEBP_LIBRARY WEBP_INCLUDE_DIR)
 
     SET(WEBP_LIBRARIES ${WEBP_LIBRARY})
-    SET(WEBP_INCLUDE_DIRS ${WEBP_INCLUDE_DIR})
+
+    if(WEBP_MUX_LIBRARY)
+        SET(WEBP_LIBRARIES ${WEBP_LIBRARIES} ${WEBP_MUX_LIBRARY} ${WEBP_DEMUX_LIBRARY})
+        SET(WEBP_INCLUDE_DIRS ${WEBP_INCLUDE_DIR})
+
+        try_compile(VALID_WEBP
+        "${OpenCV_BINARY_DIR}"
+        "${OpenCV_SOURCE_DIR}/cmake/checks/webp_test.cpp"
```

Comment:
It will be clearer to use https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html here, instead of adding a new file to the source tree.

Reply:
i found  Alexander Alekhine's note  at #25661 "check_cxx_source_compiles is not used in OpenCV code base. We prefer to use try_compile() instead." but if you want i will do requested changes.

Reply:
Thx for checking ! Forget about that then.